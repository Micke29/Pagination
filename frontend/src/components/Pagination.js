import PropTypes from 'prop-types'
import React from 'react'

const Pagination = ({ usersPerPage, totalUsers, paginate }) => {
    const pageNumbers = []

    for (let i = 1; i <= Math.ceil(totalUsers / usersPerPage); i++) {
        pageNumbers.push(i)
    }

    return (
        <nav>
            <ul className="pagination">
                {pageNumbers.map(number => (
                    <li key={number} className="page-item">
                        <a onClick={() => paginate(number)} href="!#" className="page-link">
                            {number}
                        </a>
                    </li>
                ))}
            </ul>
        </nav>
    )
}

Pagination.propTypes = {
    usersPerPage: PropTypes.number.isRequired,
    totalUsers: PropTypes.number.isRequired,
    paginate: PropTypes.func.isRequired,
}

export default Pagination