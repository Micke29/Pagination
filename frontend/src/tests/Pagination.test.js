import { expect } from 'chai'
import React from 'react'
import { shallow } from 'enzyme'

import Pagination from '../components/Pagination'

const fakeUsers = [
    {
        id: 1,
        name: "test 1"
    },
    {
        id: 2,
        name: "test 2"
    },
    {
        id: 3,
        name: "test 3"
    },
    {
        id: 4,
        name: "test 4"
    },
    {
        id: 5,
        name: "test 5"
    },
    {
        id: 6,
        name: "test 6"
    },
    {
        id: 7,
        name: "test 7"
    },
    {
        id: 8,
        name: "test 8"
    }
]

const currentPage = 1
const usersPerPage = 5

const paginate = pageNumber => currentPage = pageNumber

describe('<Pagination />', () => {
    it('has 2 link', () => {
        const wrapper = shallow(<Pagination usersPerPage={usersPerPage} totalUsers={fakeUsers.length} paginate={paginate} />)

        expect(wrapper.find('li')).to.have.length(2)
    })

    it('should be match its reference snapshot', () => {
        const wrapper = shallow(<Pagination usersPerPage={usersPerPage} totalUsers={fakeUsers.length} paginate={paginate} />)

        expect(wrapper).to.matchSnapshot()
    })
})