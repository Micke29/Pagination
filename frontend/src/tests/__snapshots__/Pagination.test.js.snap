// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should be match its reference snapshot 1`] = `
<nav>
  <ul
    className="pagination"
  >
    <li
      className="page-item"
      key="1"
    >
      <a
        className="page-link"
        href="!#"
        onClick={[Function]}
      >
        1
      </a>
    </li>
    <li
      className="page-item"
      key="2"
    >
      <a
        className="page-link"
        href="!#"
        onClick={[Function]}
      >
        2
      </a>
    </li>
  </ul>
</nav>
`;
